### ISSUE TITLE
ExampleTableViewController - editing support is dodgy

### ISSUE THREAD

**Issue by SuperWomble** [loosestar/MissTableViewNavigationController#9]
The editing support is a hodgepodge of calls, seemingly thrown in to "make it work". The result is confusing, and broken. 

Part of the problem is that Apple's documentation is so poor.

Suggestion: restart from the basic template, see what Apple provides by default, use that.

Xcode -> New File -> CocoaTouch Class -> UITableViewController subclass

Then, uncomment-out the Editing related delegate and datasource methods.

In particular, search for these line:

`// Delete the row from the data source`

` // Create a new instance of the appropriate class`

IMO, they should replace these lines with an #error directive, so the code won't compile until the lines are fixed.

**Comment by loosestar** [comment]
I agree, my reluctance is palpable, but for the greater good and completely understanding what I'm doing, I think following that plan will be worth it.
Will begin tomorrow. I can see me proceeding down a similar line if I get into it today.

**Comment by SuperWomble** [comment]
What I mean is, just create a new test project with a UITableViewController. Look at what is happening in it, use it to add and remove basic test data (numbers) without any additional UI. 

The focus should be on getting each part correct first. Then, after that is sorted out, UI for entering names (or whatever) can be added.  

Once that is done, you can fix ExampleTestViewController with what you have learned.

**Comment by SuperWomble** [comment]
FWIW, the problems you are running up against are shared by all iOS developers. Apple made some very poor decisions when they came up with these delegate patterns. They have intermingled view updating with model updating.

But they have learned some lessons. You can see that in SwiftUI, and the new iOS 13 Combine-backed data model for UITableViewController and UICollectionViewController.

Unfortunately, we still have to know how to do things the hard way for now.

**Comment by loosestar** [comment]
Getting into it now, piggy. Please forgive the slackitude. Stay tuned for exciting mail!
